src/tasks.js

```diff
@@ -0,0 +1,46 @@
+const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function addTask(description) {
+  const newTask = {
+    description,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  tasks.push(newTask);
+  saveTasks();
+}
+
+function deleteTask(index) {
+  tasks.splice(index, 1);
+  tasks.forEach((task, i) => {
+    task.index = i;
+  });
+  saveTasks();
+}
+
+function editTaskDescription(index, description) {
+  tasks[index].description = description;
+  saveTasks();
+}
+
+function clearCompletedTasks() {
+  const newTasks = tasks.filter((task) => !task.completed);
+  tasks.length = 0;
+  newTasks.forEach((task) => {
+    tasks.push(task);
+  });
+  saveTasks();
```

Comment:
The saveTasks() function is called in multiple places, which could be consolidated to reduce repetition and also The logic for updating the index property of tasks after deleting a task is repeated in both the deleteTask() and clearCompletedTasks() functions:

Reply:
Thank you a lot ](https://giphy.com/gifs/theoffice-ZfK4cXKJTTay1Ava29?utm_source=media-link&utm_medium=landing&utm_campaign=Media%20Links&utm_term=https://giphy.com/